org-code-javabuilder/playground/src/main/java/org/code/playground/Board.java

```diff
@@ -0,0 +1,101 @@
+package org.code.playground;
+
+import java.io.FileNotFoundException;
+import org.code.media.Color;
+import org.code.media.Font;
+import org.code.media.FontStyle;
+
+public class Board {
+    /**
+     * Returns the width of the board. This will always be 400.
+     * 
+     * @return the width of the board in pixels.
+     */
+    public int getWidth();
+
+    /**
+     * Returns the height of the board. This will always be 400.
+     * 
+     * @return the height of the board in pixels.
+     */
+    public int getHeight();
+
+    /**
+     * Sets the background of the board to the provided image. The image will
+     * be scaled to fit the full board, which may distort the image if
+     * it is not square.
+     * 
+     * @param filename the name of the file from the asset manager to put in the
+     *                 background
+     * @throws FileNotFoundException if the file cannot be found in the asset
+     *                               manager
+     */
+    public void setBackgroundImage(String filename) throws FileNotFoundException;
+
+    /**
+     * Adds a clickable image to the board.
+     * 
+     * @param image the image to add. If the image is already on the board,
+     *              this method does nothing.
+     */
+    public void addClickableImage(ClickableImage image);
+
+    /**
+     * Removes the clickable image from the board.
+     * 
+     * @param image the image to remove. If the image is not on the board, this
+     *             method does nothing.
+     */
+    public void removeClickableImage(ClickableImage item);
+
+    /**
+     * Adds a non-clickable item to the board. 
+     *
+     * @param item the item to add. If the item is already on the board,
+     *             this method does nothing.
+     */
+    public void addItem(Item item);
+
+    /**
+     * Removes the non-clickable item from the board.
+     * 
+     * @param item the item to remove. If the image is not on the board, this
+     *             method does nothing.
+     */
+    public void removeItem(Item item);
+    
+     /**
+     * Plays a sound from the asset manager.
+     * 
+     * @param filename the name of the sound file from the asset manager to play
+     * @throws FileNotFoundException when the sound file cannot be found.
+     */
+    public void playSound(String filename) throws FileNotFoundException;
```

Comment:
Now that we're set on the frontend approach, I realized throwing FileNotFoundExceptions get a little tricky - since we don't actually need to load the asset file in Javabuilder, we don't know if the file exists or not, but I'm guessing we want to keep these exceptions around since we expect the students to explicitly catch them (like in theater)? If so, a couple ideas:
- load the files just to check if they exist but send back filenames. not sure about this since it would just inflate load times without any performance or feature benefits
- pre-cache the list of asset file names in javabuilder and check if the given filename matches one of them. this would only make the initial load slower, but i'm also not sure how this would work if students can upload files to the asset manager while the game is running

@molly-moen thoughts?

Reply:
good call out. Another option would be for javalab to send a list of filenames on program run, which might also speed up our image/audio load times. However, we currently don't have a way to send initial data to the lambda outside of the authorizer payload, and this doesn't make a ton of sense to include there. But we could do it if need be or figure out another method. Otherwise I think the second option would also work. I feel like it might make sense to not allow files to be used that were loaded after the game started running, because in 'real Java' you can't add extra files on the fly since everything is compiled.

Reply:
Definitely agree that we should not allow the program to use assets uploaded while the program is running.